integration_test/cases/type.exs

```diff
@@ -26,11 +26,16 @@ defmodule Ecto.Integration.TypeTest do
     # Integers
     assert [1] = TestRepo.all(from p in Post, where: p.visits == ^integer, select: p.visits)
     assert [1] = TestRepo.all(from p in Post, where: p.visits == 1, select: p.visits)
+    assert [3] = TestRepo.all(from p in Post, select: p.visits + 2)
 
     # Floats
     assert [0.1] = TestRepo.all(from p in Post, where: p.intensity == ^float, select: p.intensity)
     assert [0.1] = TestRepo.all(from p in Post, where: p.intensity == 0.1, select: p.intensity)
     assert [1500.0] = TestRepo.all(from p in Post, select: 1500.0)
+    assert [1.1] = TestRepo.all(from p in Post, select: p.intensity + ^integer)
+    assert [1.1] = TestRepo.all(from p in Post, select: p.visits + 0.1)
+    # FIXME There should be no need for typecasting (?)
```

Comment:
If this is not allowed without type casting, the ideal would be to not allow mixing types. Always integer | decimal + integer | decimal or float + float.